src/vector/jitsi/index.ts

```diff
@@ -203,6 +169,43 @@ const ack = (ev: CustomEvent<IWidgetApiRequest>) => widgetApi.transport.reply(ev
                     }
                 },
             );
+
+            widgetApi.start();
```

Comment:
Just to confirm what kind of failure you mean - are you worried about events being handled before the rest of the config parsing & setup has happened, or something different?

I moved the listeners earlier, since it was possible on the react-sdk side to click 'Join' while the widget was still awaiting the config, causing your join event to never get handled and time out. Though I realize now that if I add the join listener early like this, its handler needs to wait for the setup to be finished, for that fix to actually work as intended.

Reply:
All of the other listeners should be benign if they receive an event early, since they all noop if meetApi doesn't exist yet

Reply:
The widget ends up falling completely apart if it doesn't get set up fast enough, so the code deliberately makes sure to call `widgetApi.start()` as soon as possible. Then it validates the config/query string to ensure it's able to continue loading, then binds all the listeners and such so user interactions don't cause too many issues.

We can probably add a new widget action which goes widget->client to cover the join button being active too early, similar to how the last thing the Jitsi widget setup does is enable the join button at the very last step. In that same call, we could send an update to the react-sdk to let it know we're actually ready for it.

Reply:
OK, still not sure I entirely understand why it would fall apart from something as fast as registering a listener, but I'll trust your suggestion.

Reply:
It was more the move to *after* the config.json fetch that was the concern. Registering listeners is typically benign, but paranoid.